contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,226 @@
+# REST Architectural Style
+
+Author: [Wen Xin](https://github.com/wenmogu)
+
+This book chapter introduces the REST architectural style, which is the guideline of development for the Web we use today. The following content is a summary of selected parts from the famous [Roy Fielding's paper about REST](https://www.ics.uci.edu/~fielding/pubs/dissertation/fielding_dissertation.pdf) with my own understanding, which I hope can serve as a relatively brief, easy-to-understand and at the same time detailed introduction to REST.
+
+**Requirement of Web development --- Background of REST architectural style**
+
+The forefather of Web Sir Tim Berners-Lee envisioned the Web to be &quot;a shared information space through which people and machines could communicate.&quot; Fielding&#39;s understanding of &quot;this shared information space&quot; is that the Web is a place where everyone&#39;s information is structured and stored in a way such that their own information &quot;could be usable to themselves and others&quot;, and they can &quot;reference and structure&quot; others&#39; information such that they do not need to keep a local copy of it.
+
+The design of the Web needs to consider the properties of its users and their information. The target audience of the Web are people in different organizations all over the world connected via the Internet. Their machines to access the Web might be very different, with different operating systems and requesting different file formats. These people&#39;s information also has a large variety in terms of the content type and the format. These restraints give rise to the following requirements of the Web:
```

Comment:
How does this paragraph relate to what Sir Tim Berners-Lee envisioned? It would be nice if the transition is made clearer.

Reply:
"As such, the machines people use to access the Web might be very different..."

Reply:
I believe you mean "constraint" rather than "restraint"